pkg/controller.v1beta1/suggestion/composer/composer.go

```diff
@@ -181,21 +181,30 @@ func (g *General) desiredContainers(s *suggestionsv1beta1.Suggestion,
 	suggestionConfigData katibconfig.SuggestionConfig,
 	earlyStoppingConfigData katibconfig.EarlyStoppingConfig) []corev1.Container {
 
-	containers := []corev1.Container{}
-	suggestionContainer := corev1.Container{
-		Name:            consts.ContainerSuggestion,
-		Image:           suggestionConfigData.Image,
-		ImagePullPolicy: suggestionConfigData.ImagePullPolicy,
-		Ports: []corev1.ContainerPort{
-			{
-				Name:          consts.DefaultSuggestionPortName,
-				ContainerPort: consts.DefaultSuggestionPort,
-			},
-		},
-		Resources: suggestionConfigData.Resource,
+	var (
+		containers          []corev1.Container
+		suggestionContainer corev1.Container
+	)
+
+	suggestionConfigData.Container.DeepCopyInto(&suggestionContainer)
+
+	// Assign default values for suggestionContainer fields that are not set via
+	// the suggestion config.
+
+	if suggestionContainer.Name == "" {
+		suggestionContainer.Name = consts.ContainerSuggestion
 	}
 
-	if viper.GetBool(consts.ConfigEnableGRPCProbeInSuggestion) {
+	if !containsContainerPortWithName(suggestionContainer.Ports, consts.DefaultSuggestionPortName) &&
+		!containsContainerPort(suggestionConfigData.Ports, consts.DefaultSuggestionPort) {
```

Comment:
Should we verify that port `name` and `containerPort` is set together in the Katib Config ?
[`DefaultSuggestionPortName` and `DefaultSuggestionPort` is always used for the Suggestion's service](https://github.com/kubeflow/katib/blob/master/pkg/controller.v1beta1/suggestion/composer/composer.go#L147-L151).
So if user set `DefaultSuggestionPortName` port with different address, the Suggestion service failed.

cc @tenzen-y

Reply:
cc @johnugeorge

Reply:
It might be better to prohibit setting containerPort in katibConfig.

Reply:
@fischor Do you have time to check my latest commit ? We are getting closer for the Katib 0.15 release, and it would be great to include your feature in that release.

Reply:
@andreyvelich I should have some spare time tomorrow. Do we do it as you suggested: verify that name and port are correct, otherwise return an error in `DesiredDeployment` method? Or prohibit the DefaultSuggestionPort to be set at all?

Reply:
@fischor Thank you! Let's prohibit to set `DefaultSuggestionPort` for now as @tenzen-y suggested. So we can avoid errors from the user side.

Reply:
@fischor We are planning to cut a release in a day. Do you have any update?

Reply:
@johnugeorge Thanks for the reminder...you can check the latest commit now.